pydantic/_internal/_generate_schema.py

```diff
@@ -1458,7 +1458,12 @@ def _typed_dict_schema(self, typed_dict_cls: Any, origin: Any) -> core_schema.Co
                 else:
                     field_docstrings = None
 
-                for field_name, annotation in get_cls_type_hints_lenient(typed_dict_cls).items():
+                try:
+                    annotations = get_cls_type_hints(typed_dict_cls, ns_resolver=self._ns_resolver, lenient=False)
```

Comment:
Note here that we use `lenient=False`, because if we fail to resolve them here, there's no reason to continue

Reply:
Nice, great change.